slasher/db/kv/backup.go

```diff
@@ -0,0 +1,67 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/prysmaticlabs/prysm/shared/fileutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/sirupsen/logrus"
+	bolt "go.etcd.io/bbolt"
+	"go.opencensus.io/trace"
+)
+
+const backupsDirectoryName = "backups"
+
+// Backup the database to the datadir backup directory.
+// Example for backup: $DATADIR/backups/prysm_slasherdb_10291092.backup
+func (s *Store) Backup(ctx context.Context, outputDir string) error {
+	ctx, span := trace.StartSpan(ctx, "SlasherDB.Backup")
+	defer span.End()
+
+	var backupsDir string
+	var err error
+	if outputDir != "" {
+		backupsDir, err = fileutil.ExpandPath(outputDir)
+		if err != nil {
+			return err
+		}
+	} else {
+		backupsDir = path.Join(s.databasePath, backupsDirectoryName)
+	}
+	// Ensure the backups directory exists.
+	if err := fileutil.MkdirAll(backupsDir); err != nil {
+		return err
+	}
+	backupPath := path.Join(backupsDir, fmt.Sprintf("prysm_slasherdb_%d.backup", time.Now().Unix()))
+	logrus.WithField("prefix", "db").WithField("backup", backupPath).Info("Writing backup database")
+
+	copyDB, err := bolt.Open(
```

Comment:
Why not just copy the entire file and rename it? Wouldn't this work? Are we worried about permissions?

Reply:
It wouldn't work because this webhook is being called at runtime while there is a lock on read access to the DB. If we copy the file, we get around this lock and we could be copying a corrupted database